woocommerce/woocommerce
# Add event tracking product page spotlight tour

### Changes proposed in this Pull Request:

Closes #33160.

Adds these tracks for old product walkthrough:
1. `walkthrough_product_view` when starting walkthrough
1. `walkthrough_product_completed` when either clicking dismiss or `Next` button on last step
2. `walkthrough_product_dismissed` when dismissing walkthrough in steps before the last

Adds these tracks for the new product spotlight tour:
1. `walkthrough_product_view` when starting walkthrough
2. `walkthrough_product_enable_button_click` when clicking on "Enable guided mode" button
1. `walkthrough_product_completed` when either clicking dismiss or `Keep editing` button on last step
3. `walkthrough_product_dismissed` when clicking `x` in steps before the last
1. `walkthrough_product_step_completed` when clicking on `Next` each step

### How to test the changes in this Pull Request:

**Testing tracks on old walkthrough**

1. **Without** enabling the `woocommerce_products_tour` experiment
2. Open your browser console and make sure tracking is enabled
4. Enable tracking debugging with `localStorage.setItem( 'debug', 'wc-admin:*' )`
1. Go to WooCommerce > Home > Add my products > Start with a template > Physical product
5. Observe `wcadmin_walkthrough_product_view` track with props `{ spotlight: no, product_template: physical }`
1. Click on `dismiss` button, observe `wcadmin_walkthrough_product_dismissed` with props `{ step_name: 'old-product-name' }`
2. Refresh the browser to get the walkthrough again
3. Click on `Next` until finish
4. Observe `wcadmin_walkthrough_product_completed` track

**Testing tracks on spotlight tour**

1. Install and activate [WCA Test Helper](https://github.com/woocommerce/woocommerce-admin-test-helper/releases/download/v0.7.6/woocommerce-admin-test-helper.zip)
1. Go to Tools > WCA Test Helper > Features
2. Enable `experimental-product-tour`
1. Go to Tools > WCA Test Helper > Experiments
1. Add `woocommerce_products_tour` to treatment
1. Go to WooCommerce > Home > Add my products > Start with a template > Physical product
4. Observe `wcadmin_walkthrough_product_view` track with props `{ spotlight: yes, product_template: physical }`
1. Click on `x` button, observe `wcadmin_walkthrough_product_dismissed` with props `{ step_name: 'product-name' }`
2. Click on "Enable guided mode", observe `wcadmin_walkthrough_product_enable_button_click` track
3. Click on `Next`, observe `wcadmin_walkthrough_product_step_completed` with props `{ step_name: 'product-name', added_info: 'no' }`
4. Type in something to the product description and click on `Next`
5. Observe `wcadmin_walkthrough_product_step_completed` with props `{ step_name: 'product-description', added_info: 'yes' }` 
6. Click on `Next` until finish
7. Observe `wcadmin_walkthrough_product_completed` track

### Other information:




### FOR PR REVIEWER ONLY:
